src/SprykerEco/Zed/Easycredit/Business/Api/RequestSender/RequestSender.php

```diff
@@ -0,0 +1,195 @@
+     * @param \SprykerEco\Zed\Easycredit\Persistence\EasycreditEntityManagerInterface $easycreditEntityManager
+     */
+    public function __construct(
+        MapperInterface $mapper,
+        AdapterFactoryInterface $adapterFactory,
+        ResponseParserInterface $responseParser,
+        EasycreditLoggerInterface $logger,
+        EasycreditRepositoryInterface $easycreditRepository,
+        EasycreditEntityManagerInterface $easycreditEntityManager
+    ) {
+        $this->mapper = $mapper;
+        $this->adapterFactory = $adapterFactory;
+        $this->responseParser = $responseParser;
+        $this->logger = $logger;
+        $this->easycreditRepository = $easycreditRepository;
+        $this->easycreditEntityManager = $easycreditEntityManager;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\EasycreditInitializePaymentResponseTransfer
+     */
+    public function sendInitializePaymentRequest(QuoteTransfer $quoteTransfer): EasycreditInitializePaymentResponseTransfer
+    {
+        $requestTransfer = $this->mapper->mapInitializePaymentRequest($quoteTransfer);
+        $responseTransfer = $this->adapterFactory
+            ->createInitializePaymentAdapter()
+            ->sendRequest($requestTransfer);
+
+        $this->logger->saveApiLog(EasycreditLoggerInterface::LOG_TYPE_PAYMENT_INITIALIZE, $requestTransfer, $responseTransfer);
+
+        return $this->responseParser->parseInitializePaymentResponse($responseTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\EasycreditPreContractualInformationAndRedemptionPlanResponseTransfer
+     */
+    public function sendPreContractualInformationAndRedemptionPlanRequest(QuoteTransfer $quoteTransfer): EasycreditPreContractualInformationAndRedemptionPlanResponseTransfer
+    {
+        $requestTransfer = $this->mapper->mapPreContractualInformationAndRedemptionPlanRequest($quoteTransfer);
+        $responseTransfer = $this->adapterFactory
+            ->createPreContractualInformationAndRedemptionPlanAdapter()
+            ->sendRequest($requestTransfer);
+
+        $this->logger->saveApiLog(EasycreditLoggerInterface::LOG_TYPE_PRE_CONTRACTUAL_INFORMATION_AMD_REDEMPTION_PLAN, $requestTransfer, $responseTransfer);
+
+        return $this->responseParser->parsePreContractualInformationAndRedemptionPlanResponse($responseTransfer);
+    }
+
+    /**
+     * @param int $fkSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\EasycreditOrderConfirmationResponseTransfer
+     */
+    public function sendOrderConfirmationRequest(int $fkSalesOrder): EasycreditOrderConfirmationResponseTransfer
+    {
+        $paymentEasycreditOrderIdentifierTransfer = $this->easycreditRepository->findPaymentEasycreditOrderIdentifierByFkSalesOrderItem($fkSalesOrder);
```

Comment:
Method has too high complexity and too many responsibilities. Please split into multiple.

Reply:
Done